python_modules/dagster/dagster/_core/storage/base_storage.py

```diff
@@ -31,3 +33,8 @@ def run_storage(self) -> RunStorage:
     @abstractmethod
     def schedule_storage(self) -> ScheduleStorage:
         raise NotImplementedError()
+
+    @property
+    @abstractmethod
+    def partitions_storage(self) -> Optional[PartitionsStorage]:
```

Comment:
I don't think we should implement a different storage abstraction... let's roll this into event log storage or run storage.